neurodamus/metype.py

```diff
@@ -156,7 +156,15 @@ def _instantiate_cell(self, gid, etype_path, emodel, morpho_path, meinfos_v6, de
         add_params = meinfos_v6.add_params or (keep_axon,)  # Keep axon incompatible with add_params
 
         logging.debug("Loading Gid %d: emodel: %s, Morphology: %s", gid, emodel, morpho_file)
-        self._cellref = EModel(gid, morpho_path, morpho_file, *add_params)
+        try:
+            # For this step, do not call mpi_abort in neuron and let neurodamus handle and abort,
+            # NB: Do not re-raise as ConfigurationError, neurodamus doesn't call mpi_abort so hangs
+            old_flag = Nd.pc.mpiabort_on_error(0)
+            self._cellref = EModel(gid, morpho_path, morpho_file, *add_params)
+            Nd.pc.mpiabort_on_error(old_flag)
+        except Exception as e:
+            raise Exception("Error when loading Gid %d: emodel: %s, Morphology: %s"
+                            % (gid, emodel, morpho_file)) from e
```

Comment:
Does neurodamus have an `Exception` class of its own, e.g. `NeurodamusError` or so?

Reply:
It has ConfigurationError, but for config exceptions I guess..?

Reply:
Then maybe `RuntimeError`?

Reply:
The exception coming from neuron is of type `RuntimeError`, sure I can re-raise the same type.